src/components/select/select-component.tsx

```diff
@@ -8,13 +8,24 @@ import { Icon, Text } from 'components';
 import { InputComponent } from '../../utils/types';
 import { customStyles } from './style';
 
-const CustomOption = (props: any): JSX.Element => {
-  const { children, isSelected } = props;
+const CustomOption = ({ deleteItemIcon, onDeleteItem, ...props }: any): JSX.Element => {
+  const { children, isSelected, data } = props;
 
   return (
     <components.Option {...props}>
-      {isSelected && <Icon icon={Icon.icons.check} className="icon" />}
+      {isSelected && <Icon icon={Icon.icons.check} className='icon' />}
       {children}
+      {/* eslint-disable-next-line no-underscore-dangle */}
+      {onDeleteItem && !data.__isNew__ && (
+        <Icon
+          icon={deleteItemIcon || Icon.icons.trash}
+          className='deleteItemIcon'
+          onClick={(e) => {
+            e.stopPropagation();
+            onDeleteItem(data);
+          }}
```

Comment:
add an extra function `handleDelete` for this

Reply:
done